admin/category_product_listing.php

```diff
@@ -25,1230 +25,1656 @@
+                        }
+                        ?>
+                        <?php
+                        if ($show_prod_labels || SHOW_COUNTS_ADMIN === 'true') { ?>
+                            <th class="text-right hidden-sm hidden-xs"><?php
+                                echo TABLE_HEADING_QUANTITY; ?></th>
+                        <?php
+                        } ?>
+                        <?php
+                        // -----
+                        // Additional column-headings can be added after the Quantity column.
+                        //
+                        // A watching observer can provide an associative array in the following format (for the products' listing ONLY):
+                        //
+                        // $extra_headings = array(
+                        //     array(
+                        //       'align' => $alignment,    // One of 'center', 'right', or 'left' (optional)
+                        //       'text' => $value
+                        //     ),
+                        // );
+                        //
+                        // Observer notes:
+                        // - Be sure to check that the $p2/$extra_headings value is specifically (bool)false before initializing, since
+                        //   multiple observers might be injecting content!
+                        // - If heading-columns are added, be sure to add the associated data columns, too, via the
+                        //   'NOTIFY_ADMIN_PROD_LISTING_DATA_AFTER_QTY' notification.
+                        //
+                        if ($show_prod_labels) {
+                            $extra_headings = false;
+                            $zco_notifier->notify('NOTIFY_ADMIN_PROD_LISTING_HEADERS_AFTER_QTY', '', $extra_headings);
+                            if (is_array($extra_headings)) {
+                                foreach ($extra_headings as $heading_info) {
+                                    $align = (isset($heading_info['align'])) ? (' text-' . $heading_info['align']) : '';
+                                    ?>
+                                    <th class="hidden-sm hidden-xs<?php
+                                    echo $align; ?>"><?php
+                                        echo $heading_info['text']; ?></th>
+                                    <?php
+                                }
+                            }
+                        }
+                        ?>
+                        <th class="text-right"><?php
+                            echo TABLE_HEADING_STATUS; ?></th>
+                        <?php
+                        if ($action === '') {
+                            ?>
+                            <th class="text-right hidden-sm hidden-xs"><?php
+                                echo TABLE_HEADING_CATEGORIES_SORT_ORDER; ?></th>
+                            <th class="text-right"><?php
+                                echo TABLE_HEADING_ACTION; ?></th>
```

Comment:
As before, use a single line for each <th>

Reply:
Agreed.